packages/graphback-keycloak-authz/src/KeycloakCrudService.ts

```diff
@@ -66,41 +69,116 @@ export class KeycloakCrudService<T = any> extends GraphbackProxyService<T> {
   }
 
-  public findOne(args: any, context: any): Promise<T> {
+  public findOne(args: any, context: GraphbackContext): Promise<T> {
     if (this.authConfig.read && this.authConfig.read.roles && this.authConfig.read.roles.length > 0) {
       const { roles } = this.authConfig.read;
       if (!isAuthorizedByRole(roles, context)) {
         throw new UnauthorizedError()
       }
     }
 
+    this.checkAuthRulesForSelections(context);
+
     return super.findOne(args, context);
   }
 
-  public findBy(filter: any, context: any, page?: GraphbackPage, orderBy?: GraphbackOrderBy): Promise<ResultList<T>> {
+  public findBy(filter: any, context: GraphbackContext | any, page?: GraphbackPage, orderBy?: GraphbackOrderBy): Promise<ResultList<T>> {
     if (this.authConfig.read && this.authConfig.read.roles && this.authConfig.read.roles.length > 0) {
       const { roles } = this.authConfig.read;
       if (!isAuthorizedByRole(roles, context)) {
         throw new UnauthorizedError()
       }
     }
 
+    // TODO hardcoded context
+    if (this.authConfig.filterUsingAuthInfo && context.kauth) {
+      if (typeof this.authConfig.filterUsingAuthInfo !== 'function') {
+        throw new Error("Wrong auth filter implementation")
+      }
+
+      filter = this.authConfig.filterUsingAuthInfo(filter, context.kauth.accessToken.content)
+    }
+
+    this.checkAuthRulesForSelections(context);
+
     return super.findBy(filter, context, page, orderBy);
   }
 
   public subscribeToCreate(filter?: any, context?: any): AsyncIterator<T> {
+    if (this.authConfig.subCreate && this.authConfig.subCreate.roles && this.authConfig.subCreate.roles.length > 0) {
+      const { roles } = this.authConfig.subCreate;
+      if (!isAuthorizedByRole(roles, context)) {
+        throw new UnauthorizedError()
+      }
+    }
+
     return super.subscribeToCreate(filter, context)
   }
 
   public subscribeToUpdate(filter?: any, context?: any): AsyncIterator<T> {
+    if (this.authConfig.subUpdate && this.authConfig.subUpdate.roles && this.authConfig.subUpdate.roles.length > 0) {
+      const { roles } = this.authConfig.subUpdate;
+      if (!isAuthorizedByRole(roles, context)) {
+        throw new UnauthorizedError()
+      }
+    }
+
     return super.subscribeToUpdate(filter, context)
   }
 
   public subscribeToDelete(filter?: any, context?: any): AsyncIterator<T> {
+    if (this.authConfig.subDelete && this.authConfig.subDelete.roles && this.authConfig.subDelete.roles.length > 0) {
+      const { roles } = this.authConfig.subDelete;
+      if (!isAuthorizedByRole(roles, context)) {
+        throw new UnauthorizedError()
+      }
+    }
+
     return super.subscribeToDelete(filter, context)
   }
 
-  public batchLoadData(relationField: string, id: string | number, filter: any, context: any) {
+  public batchLoadData(relationField: string, id: string | number, filter: any, context: GraphbackContext) {
+    if (this.authConfig?.relations[relationField]?.roles.length > 0) {
```

Comment:
I see that this `relations` rule is only applied here, will it also work for `oneToOne` or `manyToOne` relationship?

Reply:
Good point. It will actually require us to refactor it. We seem to do queries on resolver which is bad.

Reply:
Is there a way to solve it?

Reply:
Hmm.. I have opened issue for this and explained that this is documentation issue

#1638

Reply:
Consider the following scenarios: 

I have the following business models, showing relationships between between `One` and `Two`.

```graphql
  type One {
      id: ID
      """ @oneToOne(field: 'prev') """
      next: Two
    }

    """
    @model 
    """
    type Two {
      id: ID!
      prev: One
    }
```

I have two roles: `basic`, `advanced`.

If I have a `auth` config like this:

```ts
  const authConfig = {
      One: {
        relations: { prev: { roles: ['advanced'] } }
      }
  }
```  

With this configuration, I want the user to retrieve only the `next` only if they have advanced level. 

The typical graphql query I'll be using for that is: 
```graphql
query {
    findOnes {
        items {
            id, 
            next {
                id                
            }
        }
    }
}

```

This query will land me to this resolver https://github.com/aerogear/graphback/blob/6d9538776f64393b459401f5f4fdbaf730439c11/packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts#L697 

which will land us here https://github.com/aerogear/graphback/blob/6d9538776f64393b459401f5f4fdbaf730439c11/packages/graphback-keycloak-authz/src/KeycloakCrudService.ts#L74

which is current missing the the relations authz check. 

The solution to this could to find a way to add those checks or documentating that this could be achieved by using the https://github.com/aerogear/graphback/blob/6d9538776f64393b459401f5f4fdbaf730439c11/packages/graphback-keycloak-authz/src/KeycloakCrudService.ts#L82 (field selection on the joining column )

Reply:
Actually that is the problem. @machi1990 Let's port that to the new issue